docs/glossary.md

```diff
@@ -0,0 +1,181 @@
+introduced by a commit compared to its parent(s), and `jj rebase` will apply
+those changes onto another base commit.
+
+The word "revision" is used as a synonym for "commit".
+
+## Commit ID
+
+A commit ID is a unique identifier for a [commit](#commit). They are 20 bytes
+long when using the Git backend. They are presented in regular hexadecimal
+format at the end of the line in `jj log`, using 12 hexadecimal digits by
+default.
+
+## Co-located repos
+
+When using the Git [backend](#backend) and the backing Git repository's `.git/`
+directory is a sibling of `.jj/`, the we call the repository "co-located".
+
+See [here](git-compatibility.md#co-located-jujutsugit-repos) for details.
+
+## Conflict
+
+TODO
+
+Conflicts can occur in files, but they can also occur in branches, for example.
+
+See [here](conflicts.md) for details.
+
+## Divergent change
+
+A divergent change is a [change](#change) that has more than one visible commit.
+
+## Operation
+
+A snapshot of the [visible commits](#visible-commits) and [branches](#branches)
+at a given point in time (technically a [view object](#view)), together with
+some metadata. The metadata includes the username, hostname, timestamps, and
+pointers to the operation's parents.
+
+## Operation log
+
+The [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) formed by
+[operation](#operation) objects, much in the same way that commits form a DAG,
+which is sometimes called the "commit history".
+
+## Repository
+
+Basically everything reachable from the [operation log](#operation-log), i.e.
+the full set of [operations](#operation) and [commits](#commit).
+
+## Remote
+
+TODO
+
+## Revision
+
+A synonym for [Commit](#commit).
+
+## Revset
+
+Jujutsu supports a functional language for selecting a set of revisions.
+Expressions in this language are called "revsets". See [here](revsets.md) for
+details.
+
+## Rewrite
+
+To "rewrite" a commit means to change its contents, its metadata (including its
+parent pointers), or both. Rewriting a commit results in a new commit, and thus
+a new [commit ID](#commit-id), but the [change ID](#change-id) generally remains
+the same.
+
+## Root commit
+
+The root commit is a virtual commit at the root of every repository. It has a
+commit ID consisting of all '0's (`00000000...`) and a change ID consisting of
+all 'z's (`zzzzzzzz...`). It can be referred to in [revsets](#revset) by the
+special name `root`. Note that our definition of "root commit" is different from
+Git's, where root commits are the first commit(s) in the repository, i.e. the
+commits `jj log -r root+` will show.
+
+## Tree
+
+A tree object represents a snapshot of a directory in the repository. Tree
+objects are defined recursively; each tree object only has the files and
+directories contained directly in the directory it represents.
+
+## Visible commits
+
+Visible commits are the commits you see in `jj log -r 'all()'`. They are those
+that are reachable from a visible head in the [view](#view).
```

Comment:
I think "head"/"visible head" needs definition.

Reply:
I've tried to clarify it under #View. I actually had a "Visible heads" as its own section before but I just ended up with a circular definition and didn't know how to explain it better. Let me know if you have any suggestions.